daemon/graphdriver/overlay2/overlay.go

```diff
@@ -515,7 +515,9 @@ func (d *Driver) Remove(id string) error {
 	dir := d.dir(id)
 	lid, err := ioutil.ReadFile(path.Join(dir, "link"))
 	if err == nil {
-		if err := os.RemoveAll(path.Join(d.home, linkDir, string(lid))); err != nil {
+                if len(lid) == 0 {
+                        logrus.Errorf("when remove layer %v, link id is empty, and do not remove this link", id)
```

Comment:
`cannot remove link, link is empty for layer %v`

Reply:
I would say `"refusing to remove empty link for layer %v"`